modules/servers/openprovider/openprovider.php

```diff
@@ -0,0 +1,153 @@
+    ];
+}
+
+/**
+ * Return array to configure module
+ *
+ * @return array[]
+ */
+function openprovider_ConfigOptions(): array
+{
+    return [
+        'username' => [
+            'FriendlyName' => USERNAME_CONFIGURATION_NAME,
+            'Type' => 'text',
+            'SimpleMode' => true,
+        ],
+        'password' => [
+            'FriendlyName' => PASSWORD_CONFIGURATION_NAME,
+            'Type' => 'password',
+            'SimpleMode' => true,
+        ],
+        'license' => [
+            'FriendlyName' => LICENSE_TYPE_CONFIGURATION_NAME,
+            'Type' => 'text',
+            'SimpleMode' => true,
+        ],
+        'period' => [
+            'FriendlyName' => LICENSE_PERIOD_CONFIGURATION_NAME,
+            'Type' => 'dropdown',
+            'SimpleMode' => true,
+            'Options' => [1 => 1, 12 => 12, 24 => 24],
+        ],
+    ];
+}
+
+/**
+ * Function creates plesk license in Openprovider
+ *
+ * @param array $params available parameters from whmcs
+ *
+ * @return string
+ *
+ * @throws \Symfony\Component\Serializer\Exception\ExceptionInterface
+ */
+function openprovider_CreateAccount($params): string
+{
+    $username = $params['configoption1'];
+    $password = $params['configoption2'];
+    $licenseType = $params['configoption3'];
+    $period = $params['configoption4'] ?? 1;
+
+    $api = getPleskApi($username, $password);
+
+    if (is_null($api)) {
+        return ERROR_API_CLIENT_NOT_CONFIGURED;
+    }
+
+    $argsCreatePleskLicense = [
+        'items' => [
+            $licenseType
+        ],
+        'period' => $period,
+    ];
+
+    $createPleskLicenseResponse = makeApiCall($api, 'createPleskLicenseRequest', $argsCreatePleskLicense);
+
+    // I don't know why, but it works
+    sleep(1);
+
+    if ($createPleskLicenseResponse->getCode() != 0) {
+        return $createPleskLicenseResponse->getMessage();
+    }
+
+    $pleskLicenseKeyId = $createPleskLicenseResponse->getData()['key_id'];
+
+    $getPleskLicenseResponse = makeApiCall($api, 'retrievePleskLicenseRequest', [
+        'key_id' => $pleskLicenseKeyId
+    ]);
+
+    if ($getPleskLicenseResponse->getCode() != 0) {
+        return $getPleskLicenseResponse->getMessage();
+    }
+
+    $pleskLicense = $getPleskLicenseResponse->getData();
+
+    $licenseNumber = $pleskLicense['key_number'];
+    $activationCode = $pleskLicense['activation_code'];
+
+    try {
+        $customFieldNames = getCustomFieldNames();
+
+        $params['model']->serviceProperties->save([$customFieldNames[0] => $licenseNumber]);
+        $params['model']->serviceProperties->save([$customFieldNames[1] => $activationCode]);
+    } catch (Exception $e) {
+        return $e->getMessage();
+    }
+
+    return 'success';
+}
+
+/**
+ * Function returns HTML template for op-plesk product and product addon to display license data
+ *
+ * @param array $params available parameters from whmcs
+ *
+ * @return string
+ */
+function openprovider_ClientArea($params): string
```

Comment:
do we have some whmcs documentation? if yes please add link how is for openprovider_MetaData